event/views.py

```diff
@@ -41,3 +43,27 @@ def view_event(request, user_id):
     event_id = request.GET["id"]
     event = Event.manager.get(id=event_id)
     return render(request, 'event/event_info.html', {'event': event})
+
+
+def event_list(request, user_id):
+    events = Event.manager.all()
+    if request.method == "GET":
+        if 'Choose_filter' in request.GET:
+            filter_type = request.GET.get('Choose_filter')
+
+            if filter_type == 'Category' and 'Choose_Category' in request.GET:
+                category = request.GET.get('Choose_Category')
+                events = events.filter(category__name=category)
+            elif filter_type == 'Location' and 'Choose_Location' in request.GET:
+                location = request.GET.get('Choose_Location')
+                events = events.filter(location__name=location)
+
+    if request.user.is_authenticated:
+        user_id = request.user.id
+    else:
+        user_id = None
```

Comment:
you can do it in one line like:
```suggestion
    user_id = request.user.id if request.user.is_authenticated else None
```